msal/application.py

```diff
@@ -292,12 +292,14 @@ def acquire_token_silent(
                     })
             now = time.time()
             for entry in matches:
-                if entry["expires_on"] - now < 5*60:
+                expires_in = int(entry["expires_on"]) - now
+                if expires_in < 5*60:
                     continue  # Removal is not necessary, it will be overwritten
+                logger.debug("Cache hit an AT")
                 return {  # Mimic a real response
                     "access_token": entry["secret"],
                     "token_type": "Bearer",
-                    "expires_in": entry["expires_on"] - now,
+                    "expires_in": int(expires_in),  # OAuth2 specs defines it as int
```

Comment:
>int [](start = 34, length = 3)

it seems like expires_in is already an int? Did -now change it to something else? Should that then have been converted to an int?

Reply:
Good question! It is true that the `expires_in` in the original server-side response is an integer. But here in this code path we hit a token in the cache, and then re-calculate a new `expires_in` based on its original value and the [current time(), which happens to be a float in Python](https://docs.python.org/3/library/time.html#time.time). So we do a Forceful Unobvious Conversion to Keep the Integer Time, at the exact place where it is needed.